examples/audio/melgan_spectrogram_inversion.py

```diff
@@ -0,0 +1,619 @@
+        config.update(
+            {
+                "frame_length": self.frame_length,
+                "frame_step": self.frame_step,
+                "fft_length": self.fft_length,
+                "sampling_rate": self.sampling_rate,
+                "n_mel_channels": self.n_mel_channels,
+                "fmin": self.fmin,
+                "fmax": self.fmax,
+            }
+        )
+        return config
+
+
+"""
+The Residual Convolutional Block extensively uses dilations and has a total receptive
+field of 27 timesteps per block. The dilations must grow as a power of the `kernel_size`
+to ensure reduction of hissing noise in the output. The network proposed by the paper is
+as follows:
+
+<p align="center">
+<img src="https://i.imgur.com/gENKH91.png" alt="drawing" height="500"/>
+</p>
+"""
+
+# Creating the Residual Stack Layer
+
+
+class ResidualStack(tf.keras.layers.Layer):
+    """
+  Convolutional residual stack with weight normalization.
+
+  input: filter: int, determines filter size for the residual stack.
+	output: convolutional stack output.
+  """
+
+    def __init__(self, filters, **kwargs):
+        super().__init__(**kwargs)
+        self.filters = filters
+        self.c1 = addon_layers.WeightNormalization(
+            Conv1D(filters, 3, dilation_rate=1, padding="same"), data_init=False
+        )
+        self.c2 = addon_layers.WeightNormalization(
+            Conv1D(filters, 3, dilation_rate=1, padding="same"), data_init=False
+        )
+        self.c3 = addon_layers.WeightNormalization(
+            Conv1D(filters, 3, dilation_rate=3, padding="same"), data_init=False
+        )
+        self.c4 = addon_layers.WeightNormalization(
+            Conv1D(filters, 3, dilation_rate=1, padding="same"), data_init=False
+        )
+        self.c5 = addon_layers.WeightNormalization(
+            Conv1D(filters, 3, dilation_rate=9, padding="same"), data_init=False
+        )
+        self.c6 = addon_layers.WeightNormalization(
+            Conv1D(filters, 3, dilation_rate=1, padding="same"), data_init=False
+        )
+        self.lrelu = LeakyReLU
+
+    def call(self, input):
+        x1 = self.c1(input)
+        x2 = self.lrelu()(x1)
+        x3 = self.c2(x2)
+        x4 = x3 + input
+
+        x5 = self.lrelu()(x4)
+        x6 = self.c3(x5)
+        x7 = self.lrelu()(x6)
+        x8 = self.c4(x7)
+        x9 = x4 + x8
+
+        x10 = self.lrelu()(x9)
+        x11 = self.c5(x10)
+        x12 = self.lrelu()(x11)
+        x13 = self.c6(x12)
+        x14 = x9 + x13
+
+        return x14
+
+    def get_config(self):
+        config = super(ResidualStack, self).get_config()
+        config.update({"filters": self.filters})
+        return config
+
+
+"""
+Each Convolutional Block uses the dilations offered by the residual stack
+and upsamples the input data by the `upsampling_factor`.
+"""
+
+# Dilated Convolutional Block consisting of the Residual stack
+
+
+class ConvBlock(tf.keras.layers.Layer):
```

Comment:
Make this a Functional model building function

Reply:
Please see the response above